internal/executor/executor.go

```diff
@@ -139,6 +139,14 @@ func (executor *Executor) RunBuild() {
 		}
 		backgroundCommand.Logs.Finalize()
 	}
+	if executor.commandTo == "" {
+		// this was a separate container or the last stage of a pipe
+		// we can clean up working directory now
+		err = os.RemoveAll(workingDir)
```

Comment:
Looks like this will clean the dir after the task, but it might miss the cleaning when the task is aborted (let's say cancelled via the web interface)

Wouldn't it be better to clear the dir before the task is started (if it exists)?

Reply:
Sorry, just saw the comment. Good catch! Addressing it in https://github.com/cirruslabs/cirrus-ci-agent/pull/117